src/pgm_devroye.c

```diff
@@ -129,8 +132,9 @@ random_right_bounded_invgauss(bitgen_t* bitgen_state, parameter_t* const pr)
     }
     do {
         double y = random_standard_normal(bitgen_state);
-        double w = (pr->z + 0.5 * y * y) / pr->z2;
-        x = w - sqrt(w * w - 1 / pr->z2);
+        double y_z2 = y * y / pr->z2;  /* (y/z)^2 */
+        double w = 1. / pr->z + 0.5 * y_z2;
+        x = w - sqrt(y_z2 * (1. / pr->z + 0.25 * y_z2));  /* See GH-issue #83 */
```

Comment:
i don't understand what's going on above with the `else if` but this part makes sense to me. i can confirm the algebra. would it make sense to ensure that `x` is non-negative with something like `x=max(0, x)`?

Reply:
You're right, I could have just went with that instead of rewriting the whole expression.I updated it in [`94233c4` (#85)](https://github.com/zoj613/polyagamma/pull/85/commits/94233c4f650cb743a47fb2a228130185635c5457)